src/ImageSharp/Image/ImageBase{TPixel}.cs

```diff
@@ -37,6 +38,11 @@ public abstract class ImageBase<TPixel> : IImageBase<TPixel>
         private TPixel[] pixelBuffer;
 
         /// <summary>
+        /// The span representing the pixel buffer
+        /// </summary>
+        private Span<TPixel> span;
```

Comment:
[Span could be used as member in only in stack-only structs!](https://github.com/dotnet/corefxlab/blob/master/docs/specs/span.md#spant-will-be-stack-only)

MS is planning to provide a Roslyn analyzer to ensure this.

I think we should change the `Pixels` property to return a `Span<TPixel>` instead having this member! There are several reasons to do this:

- The "tail" of the `pixelBuffer` array is junk (extra bytes allocated by `ArrayPool`), so current `Pixels.Length` returns a random power-of-2 number unrelated to `Width*Height`
- Exposing pixel data as an array won't be possible after introducing pluggable memory management. (The source of the `Pixels` span might be unmanaged memory, an MMF buffer or anything.